var/spack/repos/builtin/packages/libdwarf/package.py

```diff
@@ -48,26 +50,49 @@ class Libdwarf(Package):
     version('20130729', '4cc5e48693f7b93b7aa0261e63c0e21d')
     version('20130207', '64b42692e947d5180e162e46c689dfbf')
     version('20130126', 'ded74a5e90edb5a12aac3c29d260c5db')
-    depends_on("libelf")
+    depends_on("elf", type='link')
 
     parallel = False
 
     def install(self, spec, prefix):
+
+        # elfutils contains a dwarf.h that conflicts with libdwarf's
+        if spec.satisfies('^elfutils'):
```

Comment:
I'm a little worried about this stuff because it modifies a dependency build.  I know why we need the workaround, but could we at least ensure that the cleanup code is executed even if an intervening error occurs?  Right now the move-back is not in a `finally` clause so it *may* not happen if something is raised between this conditional and the corresponding one below.

How about a context manager:

```python
from contextlib import contextmanager

@contextmanager
def hide_files(*file_list):
    baks = ['%s.bak' % f for f in file_list]
    for f, bak in zip(baks, file_list):
        shutil.move(f, bak)
    yield
    for f, bak in zip(baks, file_list):
        shutil.move(bak, f)
```

Obviously you'd need the error handling to be in there too, but that guarantees that the cleanup code runs no matter what.  Then, you could take the code between these two conditionals and move it into the context manager:

```python
with hide_files(join_path(spec['elfutils'].prefix.include, 'dwarf.h'):
    # stuff between conditionals
```

Thoughts?